bears/uploadTool.py

```diff
@@ -0,0 +1,81 @@
+import glob
+import os
+import re
+import shutil
+import sys
+
+sys.path.insert(0, '..')
+
+from bears.Constants import classifiers
+from bears.Constants import encoding
+from bears.Constants import imports
+from coalib.collecting.Importers import iimport_objects
+
+
+def main():
+    # create dir and set the path to the working directory
+    if not os.path.exists('upload'):
+        os.mkdir('upload')
+
+    bears_full_filenames = glob.glob('**/*Bear.py')
+
+    for bear_file_name in bears_full_filenames:
+        file_name = bear_file_name
+        bear_name = re.match(r'.+/(.*)\.py', bear_file_name).group(1)
+        bear_object = list(iimport_objects(file_name, attributes='kind',
+                                           local=True))
+        if not os.path.exists('upload/' + bear_name):
+            os.mkdir(os.path.join('upload', bear_name))
+        if not os.path.exists(os.path.join('upload', bear_name, bear_name)):
+            os.mkdir(os.path.join('upload', bear_name, bear_name))
+        if not os.path.exists(os.path.join('upload', bear_name, bear_name,
+                                           '__init__.py')):
+            init = open(os.path.join('upload', bear_name, bear_name,
+                                     '__init__.py'), 'w')
+            init.write(' ')
+            init.close()
+        shutil.copyfile(file_name, os.path.join('upload', bear_name, bear_name,
+                                                bear_name + '.py'))
+
+        with open('upload/' + bear_name + '/setup.py', "w") as setup:
+
+            # set env
+            setup.write('#!/usr/bin/env python3\n')
+
+            # write imports
+            setup.write(imports + '\n\n')
+
+            # encoding UTF-8
+            setup.write(encoding + '\n\n')
+
+            # setuptools
+            setup.write('if __name__== "__main__":\n')
+            setup.write("\tsetup(name=" + "'" + bear_name + "',\n")
+            setup.write('\t\t  version="0.1.30",\n')
+            setup.write("\t\t  description='The " +
+                        bear_name + " bear for coala (Code")
+            setup.write(" Analysis Application)',\n")
+            setup.write("\t\t  authors=" + str(bear_object[0].AUTHORS) + ",\n")
+            setup.write("\t\t  authors_emails="
+                        + str(bear_object[0].AUTHORS_EMAILS) + ",\n")
+            setup.write("\t\t  maintainers=" + str(bear_object[0].maintainers)
+                        + ",\n")
+            setup.write("\t\t  maintainers_emails="
+                        + str(bear_object[0].maintainers_emails) + ",\n")
+            setup.write("\t\t  platforms='" + str(bear_object[0].PLATFORMS)
+                        + "',\n")
+            setup.write("\t\t  license='" +
+                        str(bear_object[0].LICENSE) + "',\n")
+            setup.write('\t\t  classifiers=' + classifiers + ')')
+
+            # upload everything
+
+            os.chdir(os.path.join('upload', bear_name))
```

Comment:
Doing `.chdir()` can be quite tricky and error prone, so you might consider invoking `subprocess.Popen` with the kwarg `cwd` set to the corresponding folder